File: apps/api/src/mental_wellbeing_api/schemas/localization.py
from __future__ import annotations

from typing import Any

from pydantic import BaseModel, Field


class LocalizationLanguageItemResponse(BaseModel):
    language_code: str
    display_name: str
    native_name: str
    is_rtl: bool = False
    enabled: bool = True


class LocalizationCatalogResponse(BaseModel):
    default_language: str = "en"
    supported_languages: list[LocalizationLanguageItemResponse] = Field(default_factory=list)
    safety_copy_keys: list[str] = Field(default_factory=list)
    localized_ui_sections: list[str] = Field(default_factory=list)


class UserLanguagePreferenceResponse(BaseModel):
    user_id: str
    preferred_language: str
    content_language: str
    fallback_language: str
    supported_languages: list[str] = Field(default_factory=list)


class UserLanguagePreferenceUpdateRequest(BaseModel):
    preferred_language: str
    content_language: str | None = None
    fallback_language: str | None = None


class LocalizationRuntimeCopyResponse(BaseModel):
    language: str
    normalized_language: str
    copy: dict[str, Any] = Field(default_factory=dict)


class AdminLocalizationOverviewResponse(BaseModel):
    organization_id: str | None = None
    default_language: str = "en"
    supported_language_codes: list[str] = Field(default_factory=list)
    language_preference_breakdown: dict[str, int] = Field(default_factory=dict)
    content_language_breakdown: dict[str, int] = Field(default_factory=dict)
    fallback_language_breakdown: dict[str, int] = Field(default_factory=dict)
    care_plan_language_breakdown: dict[str, int] = Field(default_factory=dict)

File: apps/web/src/types/localization.ts
export type LocalizationLanguageItem = {
  language_code: string;
  display_name: string;
  native_name: string;
  is_rtl: boolean;
  enabled: boolean;
};

export type LocalizationCatalog = {
  default_language: string;
  supported_languages: LocalizationLanguageItem[];
  safety_copy_keys: string[];
  localized_ui_sections: string[];
};

export type LocalizationRuntimeCopy = {
  language: string;
  normalized_language: string;
  copy: Record<string, string>;
};

export type UserLanguagePreference = {
  user_id: string;
  preferred_language: string;
  content_language: string;
  fallback_language: string;
  supported_languages: string[];
};

export type UpdateUserLanguagePreferencePayload = {
  preferred_language: string;
  content_language?: string | null;
  fallback_language?: string | null;
};

export type AdminLocalizationOverview = {
  organization_id?: string | null;
  default_language: string;
  supported_language_codes: string[];
  language_preference_breakdown: Record<string, number>;
  content_language_breakdown: Record<string, number>;
  fallback_language_breakdown: Record<string, number>;
  care_plan_language_breakdown: Record<string, number>;
  recent_preference_adoption_7d: Record<string, number>;
  recent_preference_adoption_30d: Record<string, number>;
  fallback_usage_count: number;
  top_preferred_language: string | null;
  top_care_plan_language: string | null;
  alignment_alerts: string[];
  fallback_gap_alerts: string[];
};